Create V Entry (#2170)
Codecademy/docs: content/general/concepts/software-development-life-cycle/terms/v/v.md

---

> Title: 'V'

Reviewer: I think it's best to use a more explicit and clear title. Also please change the folder and filename to v-model.

Description: 'A model of software development that was based on the Waterfall model.'
Subjects:
  - 'Mobile Development'
  - 'Web Development'
Tags:
  - 'Development'
  - 'Models'
CatalogContent:
  - 'learn-how-to-code'
  - 'paths/full-stack-engineer-career-path'
---

The **V model** is a Software Development Life Cycle (SDLC)  that is based on the [Waterfall model](https://www.codecademy.com/resources/docs/general/software-development-life-cycle/waterfall) and emphasizes the Verification and Validation of the product. Each phase of this model is operated along with a level of testing.

> The V model was first put into use in the late twentieth century and has been widely used ever since. It can be divided in five main phases:

Reviewer: ```suggestion
The V-model was first implemented in the late twentieth century and has been widely used ever since. It can be divided in five main phases:
```

1. Requirements analysis
2. System design
3. Architectural design
4. Module design
5. Coding

> Except for Coding, the other four phases are linked to a distinct testing activity each. Going up from the bottom (Module design), those testing activities are, respectively:

Reviewer: ```suggestion
Except for coding, the other four phases are each linked to a distinct testing activity. Going up from the bottom (module design), those testing activities are, respectively:
```

> Except for **Coding**, the other four phases are linked to a distinct testing activity each. Going up from the bottom **(Module design)**, those testing activities are, respectively:

Reviewer 2: ```suggestion
Except for Coding, the other four phases are linked to a distinct testing activity each. Going up from the bottom (Module design), those testing activities are, respectively:
```

- Unit testing

> - Integration Testing

Reviewer: ```suggestion
- Integration testing
```

- System testing
- Acceptance testing

## Advantages

- Testing within each stage ensures each requirement is properly checked and validated, and that the product is of high quality and fulfills the requirements.

> - Errors and issues are detected at an early stages thanks to early testing activities.

Reviewer: ```suggestion
- Errors and issues can be detected at early stages of development due to an emphasis on testing early and often.
```
